cli/app.go

```diff
@@ -1327,7 +1327,7 @@ var app = &cli.App{
 							Usage: "removes binary data with file IDs in a single org and location from dataset",
 							UsageText: createUsageText("dataset data remove",
 								[]string{datasetFlagDatasetID, generalFlagOrgID, dataFlagLocationID, dataFlagFileIDs}, false),
-							// TODO(RSDK-9286) do we need to ask for og and location here?
+							// CR erodkin: we need these because currently the way the Data APIs work, if you don't pass the primary org ID then it won't work correctly and I don't think we can properly infer which one is the "correct" one. We could do something hacky! but probably better to leave this to data to fix, and flag this to reviewers in the meantime.
 							Flags: []cli.Flag{
```

Comment:
We continue to ask for and require all of org/location/machine for `RemoveDataFromDataset`. Per discussion with @dmhilly and @tahiyasalam, the behavior here will change depending on which org ID is passed, and if we pass a secondary org ID instead of the primary owning org ID then data will not be removed correctly.